samza-kv/src/main/java/org/apache/samza/storage/kv/BaseLocalStoreBackedTableDescriptor.java

```diff
@@ -36,11 +36,20 @@
  */
 abstract public class BaseLocalStoreBackedTableDescriptor<K, V, D extends BaseLocalStoreBackedTableDescriptor<K, V, D>>
     extends BaseTableDescriptor<K, V, D> {
+
+  static final public String INTERNAL_ENABLE_CHANGELOG = "internal.enable.changelog";
+  static final public String INTERNAL_CHANGELOG_STREAM = "internal.changelog.stream";
+  static final public String INTERNAL_CHANGELOG_REPLICATION_FACTOR = "internal.changelog.replication.factor";
+
   protected List<String> sideInputs;
   protected SideInputsProcessor sideInputsProcessor;
+  protected boolean enableChangelog = true;
```

Comment:
Are we taking a conscious decision of having changelog enabled by default for tables? The current behavior of user-defined stores in high-level or low level has changelog disabled by default.

Reply:
Well, this is a discrepancy already. But I think it is probably more advantageous to keep it enabled by default. We have an option to disable it and documented in the javadoc. What do you think?

Reply:
Sure, I don't have a strong opinion on that. However, what does it mean to have a changelog for in-memory stores or RocksDb that is in-memory? 
Also, refer to my comments below on both `changelogStream` and `changelogReplicationFactor` both being optional parameters.